### PR TITLE
WT-10249 Prevent tiered dhandles that are queued for work from being swept

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

**Review by sueloverso — APPROVED** [review]
LGTM

File: src/tiered/tiered_work.c [hunk 1]
```diff
@@ -90,6 +91,36 @@ __wt_tiered_push_work(WT_SESSION_IMPL *session, WT_TIERED_WORK_UNIT *entry)
     return;
 }
 
+/*
+ * __wt_tiered_push_work --
+ *     Push a work unit to the queue. Assumes it was previously returned from one of the get
+ *     functions, and it is being re-queued.
+ */
+void
+__wt_tiered_push_work(WT_SESSION_IMPL *session, WT_TIERED_WORK_UNIT *entry)
```
Comment (pmacko86) [inline src/tiered/tiered_work.c:100]: Would it make sense to rename this function to make it clear that this is for work that is being re-queued? (I'm not sure it's actually necessary, but I felt like it's at least worth the question.)

**Review by pmacko86 — APPROVED** [review]
LGTM.

File: src/tiered/tiered_work.c (hunk 1 above)
Comment (ddanderson) [reply src/tiered/tiered_work.c]: Agreed.  Renamed as `__wt_tiered_requeue_work`.

**Pull request merged**